src/rustup-cli/self_update.rs

```diff
@@ -1069,13 +1070,22 @@ pub fn update() -> Result<()> {
     }
     let setup_path = try!(prepare_update());
     if let Some(ref p) = setup_path {
-        info!("rustup updated successfully");
+        let version = try!(get_new_rustup_version(&p));
+        info!("rustup updated successfully to {}", version);
         try!(run_update(p));
     }
 
     Ok(())
 }
 
+fn get_new_rustup_version(path: &Path) -> Result<String> {
+    let raw_version = Command::new(path).arg("--version").output().unwrap().stdout;
+    let string_version = String::from_utf8(raw_version).unwrap_or(String::new());
+    let re = Regex::new(r"\d+.\d+.\d+").unwrap();
+    let capture = re.captures(&string_version).unwrap().at(0).unwrap();
+    Ok(String::from(capture))
+}
```

Comment:
Since this code is on the super-important upgrade path, let's be a little more defensive. What happens if the new executable doesn't run? Seems like it could happen. What happens if it fails? What happens if the output of `--version` changes? Seems like we might want to do that (for example returning a version like "2.0.0-rc1"). Instead of unwraps here can they return errors instead? Then have the caller convert those errors to a placeholder string like "(unknown)".

An aside, if the executable fails to run that would be a bad sign - it might be worth thinking about turning that into a hard error and failing the upgrade. But that can be a different PR.